src/cli/topical_doc.rs

```diff
@@ -0,0 +1,139 @@
+    std           std/index.html          root/std
+    core          core/index.html         root/core
+    alloc         alloc/index.html        root/core
+    KKK           std/keyword.KKK.html    root/std
+    PPP           std/primitive.PPP.html  root/std
+    MMM           std/macro.MMM.html      root/std
+
+
+    # len() == 2 not ending in ::
+    MMM           std/macro.MMM.html      root/std
+    KKK           std/keyword.KKK.html    root/std
+    PPP           std/primitive.PPP.html  root/std
+    MMM           core/macro.MMM.html     root/core
+    MMM           alloc/macro.MMM.html    root/alloc
+    # If above fail, try module
+    std::module   std/module/index.html   root/std/module
+    core::module  core/module/index.html  root/core/module
+    alloc::module alloc/module/index.html alloc/core/module
+
+    # len() == 2, ending with ::
+    std::module   std/module/index.html   root/std/module
+    core::module  core/module/index.html  root/core/module
+    alloc::module alloc/module/index.html alloc/core/module
+
+    # len() > 2
+    # search for index.html in rel_path
+    std::AAA::MMM std/AAA/MMM/index.html     root/std/AAA/MMM
+
+    # OR check if parent() dir exists and search for fn/sturct/etc
+    std::AAA::FFF std/AAA/fn.FFF9.html       root/std/AAA
+    std::AAA::SSS std/AAA/struct.SSS.html    root/std/AAA
+    core:AAA::SSS std/AAA/struct.SSS.html    root/coreAAA
+    alloc:AAA::SSS std/AAA/struct.SSS.html   root/coreAAA
+    std::AAA::TTT std/2222/trait.TTT.html    root/std/AAA
+    std::AAA::EEE std/2222/enum.EEE.html     root/std/AAA
+    std::AAA::TTT std/2222/type.TTT.html     root/std/AAA
+    std::AAA::CCC std/2222/constant.CCC.html root/std/AAA
+
+    **************************/
+
+    // topic.split.count cannot be 0
+    let subpath_os_path = match topic_vec.len() {
+        1 => match topic {
+            "std" | "core" | "alloc" => match index_html(&doc, &work_path) {
+                Some(f) => f,
+                None => no_document(doc.topic)?,
+            },
+            _ => {
+                let std = PathBuf::from("std");
+                search_path(&doc, &std, &keywords_top)?
+            }
+        },
+        2 => match index_html(&doc, &work_path) {
+            Some(f) => f,
+            None => {
+                let parent = work_path.parent().unwrap();
+                search_path(&doc, &parent, &keywords_top)?
+            }
+        },
+        _ => match index_html(&doc, &work_path) {
+            Some(f) => f,
+            None => {
+                // len > 2, guaranteed to have a parent, safe to unwrap
+                let parent = work_path.parent().unwrap();
+                search_path(&doc, &parent, &keywords_mod)?
+            }
+        },
+    };
+    // The path and filename were validated to be existing on the filesystem.
+    // It should be safe to unwrap, or worth panicking.
+    Ok(subpath_os_path.into_os_string().into_string().unwrap())
```

Comment:
I'd still be tempted to say you should be returning `Result<PathBuf>` so that you don't need to do these conversions.